src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocForbiddenFragmentRelativeToPeriod.java

```diff
@@ -0,0 +1,35 @@
+/*
+SummaryJavadoc
+violateExecutionOnNonTightHtml = (default)false
+forbiddenSummaryFragments = ^$|fail-summary-fragment
+period = _
```

Comment:
I notice the Japanese period looks like it has a space after it but it's not actually a separate character. Will the Japanese period have a space after it when it's really used, or will the next character be the start of the next word?

Reply:
Oh dear, it looks like because the Japanese period comes with a half-space in the same character, it doesn't need a space after it. I'm not sure how to handle this and still fix the bug with western version numbers:
- https://www.tofugu.com/japanese/japanese-punctuation/
- https://github.com/checkstyle/checkstyle/issues/14750

Reply:
I suppose I'll need to use a custom period that doesn't include whitespace in this example, and make a separate example to test that a period character that includes whitespace counts as ending the sentence without whitespace after it. Otherwise we can only support western or eastern periods and not both. I hope it's possible to make this work.

Reply:
It looks like there's a tool built into Java for detecting sentence boundaries:

https://docs.oracle.com/javase//tutorial/i18n/text/boundaryintro.html

That wants to know the locale rather than setting a custom period. It looks like it can do a lot even without knowing the locale, using Locale.ROOT. It looks like we should probably use that? I'm not sure what to do about the custom period configuration vs locale configuration though.

Reply:
> looks like there's a tool built into Java for detecting sentence boundaries:

good idea , but we need to try it first, lets do this in separate issue/PR.
let focus on making this PR to merge, it is growing with nuances.
we should not fix all possible problems in this PR, as soon we incrementally imrpove and not cause new problems (false positives), we can merge.

Reply:
I higly recommend to merge second commit in separate PR, it will be instant merge, so it will keep this PR smaller and let us focus on details more.

Reply:
That all sounds sensible to me. I'll sort out the whitespace handling.

Reply:
I've split out the second commit:
- https://github.com/checkstyle/checkstyle/pull/14774